### DISCUSSION TITLE
The meaning behind "Note" section regarding symmetric R in least squares doc

### DISCUSSION THREAD

**Discussion by ogencoglu** [qpsolvers/qpsolvers#275] [Q&A] [answered]
Least squares solver doc has the section:

> In least squares, the matrix 
 should be symmetric. Many solvers (including CVXOPT, OSQP and quadprog) leverage this property and may return unintended results when it is not the case. You can set project 
 on its symmetric part by:

`R = 0.5 * (R + R.transpose())`

How does this work in practice where R is a numpy array of shape (`n_observations` x `n_variables`) and is not a square matrix? Is it a common practice to trail bunch of zeros to make this work?

**Comment by stephane-caron** [comment, marked as answer]
That was an incorrect copy-paste from `solve_qp` :see_no_evil: Thank you for pointing it out, https://github.com/qpsolvers/qpsolvers/pull/277 fixed this note.

(The original note was for [`solve_qp`](https://qpsolvers.github.io/qpsolvers/quadratic-programming.html#qpsolvers.solve_qp), where the cost matrix $P$ is assumed to be symmetric. This condition will be met by construction when [converting from least squares to QP](https://scaron.info/blog/conversion-from-least-squares-to-quadratic-programming.html).)

**Reply by ogencoglu** [reply]
Thanks for swift reply!
